File: 0791-custom-sort-string/0791-custom-sort-string.cs
public class Solution {
    public string CustomSortString(string t, string s) 
    {
        Dictionary<char, int> mp = new Dictionary<char, int>();
        foreach (char c in s) {
            if (mp.ContainsKey(c))
                mp[c]++;
            else
                mp[c] = 1;
        }
        
        string ans = "";
        
        foreach (char c in t) {
            if (mp.ContainsKey(c) && mp[c] > 0) {
                ans += new string(c, mp[c]);
                mp[c] = -1;
            }
        }
        
        foreach (char c in s) {
            if (mp.ContainsKey(c) && mp[c] != -1)
                ans += c;
        }
        
        return ans;
    }
}

File: 0791-custom-sort-string/0791-custom-sort-string.cpp
class Solution {
public:
    string customSortString(string t, string s) 
    {
        map <char, int> mp;
        for (auto i : s) mp[i]++;
        string ans;
        for (int i = 0; i < t.size(); i++)
        {
            if (mp[t[i]])
            {
                ans += string(mp[t[i]], t[i]);
                mp[t[i]] = -1;
            }
        }
        for (int i = 0; i < s.size(); i++)
        {
            if (mp[s[i]] != -1)  ans += s[i];
        }
        return ans;
    }
};
